spec.md

```diff
@@ -1459,6 +1477,21 @@ The `else`{.firrtl} branch may also be added to the single line:
 when c : a <= b else : e <= f
 ```
 
+### Match Statements
+
+Match statements are used to discriminate the active variant of a enumeration
+typed expression.  A match statement must exhaustively test every variant of an
+enumeration. An optional binder may be specified to extract the data of the
+variant.
+
+``` firrtl
+match x:
+  #Some(x):
```

Comment:
What about reusing the existing keyword "with" instead of "#"?

Reply:
We don't really need a keyword or anything here at all, e.g. it could just be:
```firrtl
match x:
  Some(x):
```

I added the `#` for mainly for enum create expressions, e.g. `#a : [#a, #b, #c]`, and kept it in the match statements for consistency.

Reply:
sorry are Some and None example enumerations or keywords themselves here?

Reply:
if they are just examples can we stick with the `some` and `none` capitalization throughout?

Reply:
example enumerations

Reply:
is the `match x... some(x)` using the same `x` meaningful? If not, could we use different variables for the two usages?